test/extension-tests/successful-build/test/prefered-generators.test.ts

```diff
@@ -0,0 +1,213 @@
+import * as chai from 'chai';
+import {expect} from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+chai.use(chaiAsPromised);
+
+import {clearExistingKitConfigurationFile} from '../../../test-helpers';
+import {DefaultEnvironment} from '../../../helpers/test/default-environment';
+
+import {CMakeTools} from '../../../../src/cmake-tools';
+import {ITestCallbackContext, IHookCallbackContext} from 'mocha';
+
+interface BuildSystemConfiguration {
+  defaultKit: string;
+  expectedDefaultGenerator: string;
+  path?: string;
+}
+
+let workername = process.env.APPVEYOR_BUILD_WORKER_IMAGE;
+
+if (process.env.TRAVIS_OS_NAME) {
+  workername = process.env.TRAVIS_OS_NAME;
+}
+
+if (workername === undefined) {
+  workername = 'DevPC';
+}
+
+const DefaultCompilerMakeSystem: {[os: string]: BuildSystemConfiguration[]} = {
```

Comment:
`DefaultCompilerMakeSystem` is an odd name for this structure. Try `KITS_BY_PLATFORM`

Reply:
Done